src/types/any.rs

```diff
@@ -259,8 +259,10 @@ impl PyAny {
     ///
     /// # fn main() -> PyResult<()> {
     /// Python::with_gil(|py| -> PyResult<()> {
-    ///     let a: &PyInt = 0_u8.into_py(py).into_ref(py).downcast()?;
-    ///     let b: &PyInt = 42_u8.into_py(py).into_ref(py).downcast()?;
+    ///     let a = 0_u8.into_py(py).into_bound(py);
+    ///     let b = 42_u8.into_py(py).into_bound(py);
+    ///     let a: &Bound<'_, PyInt> = a.downcast()?;
+    ///     let b: &Bound<'_, PyInt> = b.downcast()?;
```

Comment:
Maybe this captures the original intent better:

```suggestion
    ///     let a: Bound<'_, PyInt> = 0_u8.into_py(py).into_bound(py).downcast_into();
    ///     let b: Bound<'_, PyInt> = 42_u8.into_py(py).into_bound(py).downcast_into();
```